fvm/meter/interaction_meter.go

```diff
@@ -0,0 +1,129 @@
+package meter
+
+import (
+	"math"
+
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type MeteredStorageInteractionMap map[StorageInteractionKey]uint64
+
+type StorageInteractionKey struct {
+	Owner, Key string
+}
+
+type InteractionMeterParameters struct {
+	storageInteractionLimit uint64
+}
+
+func DefaultInteractionMeterParameters() InteractionMeterParameters {
+	return InteractionMeterParameters{
+		storageInteractionLimit: math.MaxUint64,
+	}
+}
+
+func (params MeterParameters) WithStorageInteractionLimit(
+	maxStorageInteractionLimit uint64,
+) MeterParameters {
+	newParams := params
+	newParams.storageInteractionLimit = maxStorageInteractionLimit
+	return newParams
+}
+
+type InteractionMeter struct {
+	params InteractionMeterParameters
+
+	storageUpdateSizeMap     map[StorageInteractionKey]uint64
+	totalStorageBytesRead    uint64
+	totalStorageBytesWritten uint64
+}
+
+func NewInteractionMeter(params InteractionMeterParameters) *InteractionMeter {
+	return &InteractionMeter{
+		params:               params,
+		storageUpdateSizeMap: make(map[StorageInteractionKey]uint64),
+	}
+}
+
+// MeterStorageRead captures storage read bytes count and returns an error
+// if it goes beyond the total interaction limit and limit is enforced
+func (m *InteractionMeter) MeterStorageRead(
+	storageKey StorageInteractionKey,
+	value flow.RegisterValue,
+	enforceLimit bool) error {
```

Comment:
nit: could you add some newlines / reformat the method signatures while you're at it? (same for the methods below)